File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.cpp
// Definition for a Node.
/*
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

#pragma GCC optimize("03", "unroll-loops");
static const int __ = [](){
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    return 0;
}();

class Solution {
private:
    void traverse(Node* root, vector<int>& ans) {
        if(!root) return;
        for(const auto &child : root -> children) {
            traverse(child, ans);
        }
        ans.push_back(root -> val);
    }
public:
    vector<int> postorder(Node* root) {
        vector<int> ans;
        traverse(root, ans);
        return ans;
    }
};

File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

func postorder(root *Node) []int {
    if root == nil {
        return nil
    }
    
    var stack []*Node
    var ans []int

    stack = append(stack, root)
    for len(stack) > 0 {
        curr := stack[len(stack)-1]
        stack = stack[:len(stack)-1]
        
        ans = append(ans, curr.Val)
        
        for _, child := range curr.Children {
            stack = append(stack, child)
        }
    }
    
    for i, j := 0, len(ans)-1; i < j; i, j = i+1, j-1 {
        ans[i], ans[j] = ans[j], ans[i]
    }
    
    return ans
}